frontend-service/src/components/authView.tsx

```diff
@@ -45,20 +58,42 @@ const AuthView = ({
                 }
                 sx={{ maxWidth: '40rem' }}
             >
-                <CardContent className="grid grid-cols-2 gap-4">
-                    {
-                        formData.map(({
-                            label,
-                            type,
-                            className
-                        }) => (
-                            <div className={className}>
-                                <label className="text-start text-blue-500 font-medium">{label}</label>
-                                <Input type={type} placeholder={label} />
-                            </div>
-                        ))
-                    }
-                    <Button className="mx-auto col-span-2 text-white bg-blue-500">Submit</Button>
+                <CardContent>
+                    <Formik
+                        initialValues={ initialValues }
+                        validationSchema={ validationSchema }
+                        onSubmit={ handleSubmit }
+                    >
+                    {({ isSubmitting }) => (
+                            <Form className="grid grid-cols-2 gap-4">
+                                {formData.map(({ label, name, type, className }) => (
+                                    <div key={name} className={className}>
+                                        <label className="text-start text-blue-500 font-medium">{label}</label>
```

Comment:
Can you make a common component for label, field and Error message and use that component all over